docs/tutorial.md

```diff
@@ -0,0 +1,274 @@
+});
+```
+
+Say hi to the chat after you spawn in, using [`spawn event`](http://mineflayer.prismarine.js.org/#/api?id=quotspawnquot) and [`chat method`](http://mineflayer.prismarine.js.org/#/api?id=botchatmessage).
+
+```js
+// ...
+
+bot.once("spawn", function () {
+  bot.chat("hi!");
+  // `chat` method for sending message or commad to the chat.
+});
+```
+
+For full methods list see the documentation at [API-Bot-methods](http://mineflayer.prismarine.js.org/#/api?id=methods)
+
+### Listening for an event
+
+Bot object has many useful built in [`events`](http://mineflayer.prismarine.js.org/#/api?id=events).
+You can listen for an event by using either `on` method or `once` method of the Bot object, wich takes the name of event and a function.
+
+[`Chest`](http://mineflayer.prismarine.js.org/#/api?id=mineflayerchest), [`Furnace`](http://mineflayer.prismarine.js.org/#/api?id=mineflayerfurnace), [`Dispenser`](http://mineflayer.prismarine.js.org/#/api?id=mineflayerdispenser), [`EnchantmentTable`](http://mineflayer.prismarine.js.org/#/api?id=mineflayerenchantmenttable), [`Villager`](http://mineflayer.prismarine.js.org/#/api?id=mineflayervillager) object also has its own events.
+
+Examples :
+
+#### Echo Bot
+
+In here we're listening [`chat`](http://mineflayer.prismarine.js.org/#/api?id=quotchatquot-username-message-translate-jsonmsg-matches) event, and resend the message.
+`Chat` event only emitted when a player chats publicly.
+
+```js
+var mineflayer = require("mineflayer");
+var bot = mineflayer.createBot({...})
+
+bot.on('chat', function(username, message) {
+  if (username === bot.username) return;
+  bot.chat(message);
+});
+```
+
+Or more readable aproach,
+
+```js
+var mineflayer = require("mineflayer");
+var bot = mineflayer.createBot({...})
+
+function echo(username, message) {
+  if (username === bot.username) return;
+  bot.chat(message);
+}
+
+bot.on("chat", echo); // Only emitted when a player chats publicly.
+```
+
+`bot.on("chat", echo);` will listend for `chat` event, and execute `echo` function with argument of (username, message, translate, jsonMsg, matches) when `chat` event triggered. Read more about `chat` event [here](http://mineflayer.prismarine.js.org/#/api?id=quotchatquot-username-message-translate-jsonmsg-matches).
+
+#### Welcome Bot
+
+In here we're listening [`playerJoined`](http://mineflayer.prismarine.js.org/#/api?id=quotplayerjoinedquot-player) event.
+
+```js
+var mineflayer = require("mineflayer");
+var bot = mineflayer.createBot({...})
+
+function joined(player) {
+  bot.chat(`Welcome ${player}! :D`)
+}
+bot.on("playerJoined", joined)
+```
+
+### Callbacks
+
+Callback is a function as argument to the main function, when the main function finish or got an error then the callback executed.
+In [the API page](http://mineflayer.prismarine.js.org/#/api) callback abbreviated as `cb`.
+Callback usually takes single parameter called `err`, if `err` populated then error must be happened in the main function.
```

Comment:
The first parameter is `err` and the second (if specified in) is a value that is passed to the callback. Point that out.